src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -1479,28 +1547,217 @@ impl SimulationElement for A320BrakingForce {
+        if self.state == DoorControlState::NoControl {
+            self.duration_in_no_control += context.delta();
+        } else {
+            self.duration_in_no_control = Duration::from_secs(0);
+        }
+
+        if self.state == DoorControlState::HydControl {
+            self.duration_in_hyd_control += context.delta();
+        } else {
+            self.duration_in_hyd_control = Duration::from_secs(0);
+        }
+    }
+
+    fn update_actions_from_state(&mut self) {
+        match self.state {
+            DoorControlState::DownLocked => {}
+            DoorControlState::NoControl => {
+                self.should_close_valves = true;
+            }
+            DoorControlState::HydControl => {
+                self.should_close_valves = false;
+                if self.position_requested > Ratio::new::<ratio>(0.)
+                    || self.duration_in_hyd_control < Self::UP_CONTROL_TIME_BEFORE_DOWN_CONTROL
+                {
+                    self.control_position_request = Ratio::new::<ratio>(1.);
+                } else {
+                    self.control_position_request = Ratio::new::<ratio>(0.);
+                }
+            }
+            DoorControlState::UpLocked => {
+                self.should_close_valves = true;
+            }
+        }
+    }
+
+    fn determine_mode(
+        &mut self,
+        door: &CargoDoor,
+        pressure_available: Pressure,
+    ) -> DoorControlState {
+        match self.state {
+            DoorControlState::DownLocked => {
+                if self.position_requested > Ratio::new::<ratio>(0.) {
+                    self.should_unlock = true;
+                    DoorControlState::NoControl
+                } else {
+                    DoorControlState::DownLocked
+                }
+            }
+            DoorControlState::NoControl => {
+                if self.duration_in_no_control > Self::DELAY_UNLOCK_TO_HYDRAULIC_CONTROL {
+                    DoorControlState::HydControl
+                } else {
+                    DoorControlState::NoControl
+                }
+            }
+            DoorControlState::HydControl => {
+                self.should_unlock = false;
+                if door.is_locked() {
+                    DoorControlState::DownLocked
+                } else if door.position() > Ratio::new::<ratio>(0.9)
+                    && self.position_requested > Ratio::new::<ratio>(0.5)
+                {
+                    DoorControlState::UpLocked
+                } else {
+                    DoorControlState::HydControl
+                }
+            }
+            DoorControlState::UpLocked => {
+                if self.position_requested < Ratio::new::<ratio>(1.)
+                    && pressure_available > Pressure::new::<psi>(1000.)
+                {
+                    DoorControlState::HydControl
+                } else {
+                    DoorControlState::UpLocked
+                }
+            }
```

Comment:
**suggestion**: Consider using match guards to reduce the duplication requiring an else branch returning the same state as before:

```suggestion
        match self.state {
            DoorControlState::DownLocked if self.position_requested > Ratio::new::<ratio>(0.) => {
                self.should_unlock = true;
                DoorControlState::NoControl
            }
            DoorControlState::NoControl
                if self.duration_in_no_control > Self::DELAY_UNLOCK_TO_HYDRAULIC_CONTROL =>
            {
                DoorControlState::HydControl
            }
            DoorControlState::HydControl if door.is_locked() => {
                self.should_unlock = false;
                DoorControlState::DownLocked
            }
            DoorControlState::HydControl
                if door.position() > Ratio::new::<ratio>(0.9)
                    && self.position_requested > Ratio::new::<ratio>(0.5) =>
            {
                self.should_unlock = false;
                DoorControlState::UpLocked
            }
            DoorControlState::UpLocked
                if self.position_requested < Ratio::new::<ratio>(1.)
                    && pressure_available > Pressure::new::<psi>(1000.) =>
            {
                DoorControlState::HydControl
            }
            _ => self.state,
        }
```